lib/Immutable.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const util = require('util');
+
+/**
+ * Error for Immutable variables
+ *
+ * @extends Error
+ * @class Errors.Immutable
+ * */
+class Immutable extends Error{
+    /**
+ * @example
+    function populateUser() {
+        let user = {
+            name: 'Mike',
+            age: 99,
+        }
+        if (typeof user.age != 'undefined') {
+            const err = new Errors.Immutable;
+            err.setMessage(
+                'user.age',
+                'pupulateUser'
+            );
+            throw err;
+        }
+        user.age = 45;
+    }
+
+ populateUser();
+ * @example
+ *
+ * ```sh
+ *
+ /git/node-error-classes/example/immutable.js:16
+    throw err;
+         ^
+
+ Immutable: 'user.age' has been defined and cannot be changed on the scope of : 'pupulateUser'.
+    Variable names here must be unique
+
+ * ```
+ *
+ * @method Errors.Immutable.setMessage
+ * @param {String} varaibleName Name of variable
+ * @param {Any} scope     Scope of varaible
+ * @returns {String}    compiled error message
+ */
+    setMessage(varaibleName, scope){
+        this.name = 'Immutable';
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected indentation of 6 space characters but found 8.](https://www.codacy.com/app/RIAEvangelist/node-error-classes/file/1941079322/issues/source?bid=3073169&fileBranchId=3104106#l49)